ehrql/file_formats/__init__.py

```diff
@@ -91,6 +98,22 @@ def get_extension_from_directory(filename):
         return list(matching)[0]
 
 
+def split_directory_and_extension(filename):
+    # This is slightly unpleasant in that we've invented our own syntax for saying "I
+    # want a directory here with files of this type in it" e.g.
+    #
+    #     path/to/my/directory:csv
+    #
+    # But I don't seen an obvious alternative and I hope it's reasonably natural.
```

Comment:
Looks reasonable.  Did you consider requiring the user to pass in the file extension as a separate flag?

Reply:
Yeah, I did. It would involve adding an extra flag everywhere we want to specify outputs though, and then threading the value through to where it's needed.

As I'm planning to extend this to `generate-measures` for one-measure-per-file outputs, and to `generate-dataset` for event-level outputs, I felt it would get messy pretty quickly. There's something very nice about maintaining the simplicity of a single string which specifies where you want the output to go and what format you want it in.